make/compiler_flags

```diff
@@ -161,6 +161,22 @@ ifeq ($(OS),Windows_NT)
     CXXFLAGS_OS ?= -m64
   endif
 
+  ifeq (,$(wildcard $(MATH)make/ucrt))
+    UCRT_STRING := $(shell echo "#include <windows.h>" | $(CXX) -E -dM -  | findstr _UCRT)
+    ifneq (,$(UCRT_STRING))
+      IS_UCRT ?= true
+    else
+      IS_UCRT ?= false
+    endif
+    $(shell echo "IS_UCRT ?= $(IS_UCRT)" > $(MATH)make/ucrt)
+  else
+    -include $(MATH)make/ucrt
+  endif
```

Comment:
I think you can make this more concise by defining a make rule who's result is `make/ucrt`, which will then be built when you `include` it. This is how the `.d` files for dependencies work

Reply:
Oh cool, TIL

Reply:
Tried out the make rule, but I'm not sure about the best place to put the rule. Thoughts/prefs?

Reply:
I think putting it just above where it is included is probably fine? I think you can put rules in places that are conditionally defined